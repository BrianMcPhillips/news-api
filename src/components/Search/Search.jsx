import React from 'react';
import PropTypes from 'prop-types';

const Search = ({ searchTerm, onChange, onSubmit }) => (
  <div>
    <form onSubmit={onSubmit}>
      <label htmlFor="input">Search News Articles</label>
      <input 
        id="input"
        type="text"
        name="input"
        value={searchTerm} 
        onChange={onChange} 
      />
      <button>Find Articles</button>
    </form>
  </div>
);

Search.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Search;
