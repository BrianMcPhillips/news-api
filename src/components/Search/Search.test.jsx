import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Search from './Search';

describe('Search component', () => {
  afterEach(() => cleanup());
  it('renders search component', () => {
    const { asFragment } = render(<Search 
      searchTerm="bitcoin"
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
