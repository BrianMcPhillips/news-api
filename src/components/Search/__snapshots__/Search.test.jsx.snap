// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders search component 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="input"
      >
        Search News Articles
      </label>
      <input
        id="input"
        name="input"
        type="text"
        value="bitcoin"
      />
      <button
        data-testid="search-button"
      >
        Find Articles
      </button>
    </form>
  </div>
</DocumentFragment>
`;
